test/SuperstateTokenTestBase.t.sol

```diff
@@ -41,24 +46,32 @@ contract SuperstateTokenTestBase is Test {
     bytes32 internal constant AUTHORIZATION_TYPEHASH =
         keccak256("Permit(address owner,address spender,uint256 value,uint256 nonce,uint256 deadline)");
 
+    function getAdminAddress(address proxy) internal view returns (address) {
+        address CHEATCODE_ADDRESS = 0x7109709ECfa91a80626fF3989D68f67F5b1DD12D;
+        Vm vm = Vm(CHEATCODE_ADDRESS);
+
+        bytes32 adminSlot = vm.load(proxy, ERC1967Utils.ADMIN_SLOT);
+        return address(uint160(uint256(adminSlot)));
+    }
+
     function setUp() public virtual {
         eve = vm.addr(evePrivateKey);
 
         AllowList permsImplementation = new AllowList(address(this));
 
-        // deploy proxy admin contract
-        proxyAdmin = new ProxyAdmin();
-
         // deploy proxy contract and point it to implementation
-        permsProxy = new TransparentUpgradeableProxy(address(permsImplementation), address(proxyAdmin), "");
+        permsProxy = new TransparentUpgradeableProxy(address(permsImplementation), address(this), "");
```

Comment:
We may want to make this base `abstract` and have the implementing tests initialize the proxy.

That way the `v1` tests can use OZv4, and the `v2` tests can use OZv5

Reply:
Cool if I TODO this for later?

Reply:
Added the TODO

Reply:
👍 should do soon though, as we're going thru a lot of trouble to maintain v1 so it should be accurate